src/components/SignUp.js

```diff
@@ -1,11 +1,122 @@
-const SignUp = () => {
+import { Link, useHistory } from 'react-router-dom'
+import { useState } from 'react'
+import { FaCameraRetro } from 'react-icons/fa'
+import signupImg1 from '../assets/signupImg1.jpg'
+
+const SignUp = ({ onSignup }) => {
+    let history = useHistory()
+    const [username, setUsername] = useState('')
+    const [password, setPassword] = useState('')
+    const [first_name, setFirstName] = useState('')
+    const [last_name, setLastName] = useState('')
+    const [email, setEmail] = useState('')
+    const [admin, setAdmin] = useState(false)
+    const [errorUserName, setErrorUsername] = useState(false)
+    const [errorInputBlank, setErrorInput] = useState(false)
+
+    const onSubmit = (e) => {
+        e.preventDefault()
+        // removes any previous error from screen when successful login to dashboard
+        setErrorInput(false)
+        setErrorUsername(false)
+
+        // username taken 
+        if (username == username) {
```

Comment:
What is this checking for exactly? username will always be equal to itself. If you are checking if a particular username is taken in the database there might be different logic needed here

Reply:
you're right, thank you for pointing that out. I have altered this to `if(username)` 
This works when I tested it and is the opposite of what Jennifer is checking on the login page so I feel like it makes sense as for checking if the username already exists.